functions/index.html

```diff
@@ -0,0 +1,201 @@
+    } from "https://www.gstatic.com/firebasejs/9.7.0/firebase-database.js";
+
+    //ik weet dat er "apikey" staat maar dit hoort hier toch, is iets van firebase.
+    const firebaseConfig = {
+      apiKey: "AIzaSyCzNQ2O8_KEF7rcupBT8gNjB0_BIE7K4ig",
+      authDomain: "babble-d6ef3.firebaseapp.com",
+      databaseURL:
+        "https://babble-d6ef3-default-rtdb.europe-west1.firebasedatabase.app",
+      projectId: "babble-d6ef3",
+      storageBucket: "babble-d6ef3.appspot.com",
+      messagingSenderId: "608727980458",
+      appId: "1:608727980458:web:02a744b7927fc8b12133d3",
+      measurementId: "G-K7MPHMQNH2",
+    };
+
+    const app = initializeApp(firebaseConfig);
+    const DB = getDatabase();
+
+    let fullAnimation = anime.timeline({
+      easing: "cubicBezier(.62,.01,.7,.72)",
+      duration: 16000,
+      autoplay: false,
+    });
+
+    fullAnimation
+      .add({
+        targets: ".raffle",
+        translateX: 1500,
+        rotate: 360,
+        duration: 3000,
+      })
+      .add(
+        {
+          targets: ".raffle",
+          translateX: -1500,
+          rotate: -360,
+          duration: 3000,
+        },
+        13000
+      );
+
+    const startRaffle = () => {
+      fullAnimation.play();
+      return "Done";
+    };
+
+    const params = new URLSearchParams(window.location.search);
+    const id = params.get("id");
+    const eventRef = ref(DB, id);
+
+    onValue(eventRef, (snapshot) => {
+      const data = snapshot.val();
+      const eventType = data.type;
+      switch (eventType) {
+        case "start":
+          console.log("startcase");
+          break;
+        case "end":
+          RaffleResults(data.winners);
+          console.log("endcase");
+          break;
+        case "idle":
+          console.log("idlecase");
+          break;
+      }
+    });
+    // const evtSource = new EventSource(
+    //   `http://localhost:5001/babble-d6ef3/europe-west1/default/api/v1/raffle/listen?id=${id}`,
+    //   { withCredentials: true }
+    // );
+
+    const dummyarray = ["Osjesleben", "Blamant"];
+
+    const RaffleResults = (winnaars) => {
+      let x = document.querySelector("#raffle-winners");
+      let winnerstr = "";
+      if (winnaars.length > 8) {
+        document.querySelector("h2").style.fontSize = "2em";
+      } else {
+        document.querySelector("h2").style.fontSize = "3em";
+      }
+      for (let i = 0; i < winnaars.length; i++) {
+        winnerstr = winnerstr + winnaars[i] + "<br>";
+        // console.log(winnerstr);
+      }
+      x.innerHTML = winnerstr;
+      // startRaffle();
+      return startRaffle();
+    };
+
+    // RaffleResults(dummyarray);
+
+    // evtSource.addEventListener("end", (event) => {
+    //   const data = event.data;
+    //   console.log(data);
+    //   // console.log("test");
+    //   RaffleResults(data);
+    // });
+
+    // evtSource.onmessage = function(event) {
+    // 	let data = event.data;
+    // 	console.log(data);
+    // }
```

Comment:
weet niet precies wat al deze comments doen, maar of je moet het nog laten werken voordat het naar de alpha gaat, of ze moeten weg als je ze niet meer gebruikt

Reply:
zijn nu weg